src/cli_util.rs

```diff
@@ -1394,45 +1404,22 @@ pub fn write_commit_summary(
     Ok(())
 }
 
-pub fn write_config_entry(
-    ui: &mut Ui,
-    path: &str,
-    value: config::Value,
-) -> Result<(), CommandError> {
-    match value.kind {
-        // Handle table values specially to render each child nicely on its own line.
-        config::ValueKind::Table(table) => {
-            // TODO: Remove sorting when config crate maintains deterministic ordering.
-            for (key, table_val) in table.into_iter().sorted_by_key(|(k, _)| k.to_owned()) {
-                let key_path = match path {
-                    "" => key,
-                    _ => format!("{path}.{key}"),
-                };
-                write_config_entry(ui, key_path.as_str(), table_val)?;
-            }
-        }
-        _ => writeln!(ui, "{path}={}", serialize_config_value(value))?,
-    };
-    Ok(())
-}
-
 // TODO: Use a proper TOML library to serialize instead.
-fn serialize_config_value(value: config::Value) -> String {
+pub fn serialize_config_value(value: &config::Value) -> String {
     match value.kind {
-        config::ValueKind::Table(table) => format!(
+        config::ValueKind::Table(ref table) => format!(
```

Comment:
In a single config I think arrays of tables are extended just like any other array, not merged or anything.

But since table values can be overridden at any level I think we need to recurse over the nested tables and give each value a place to track its source & override state.

Reply:
Because the caller flattens nested tables to filter out `source == Default`, `serialize_config_value()` wouldn't receive a value of `ValueKind::Table`. This turned out to be wrong, but made me think if we need to test the source of the table items in an array.

Anyway, an array of tables in `~/.jjconfig.toml` appears to be overridden by the one in `.jj/repo/config.toml`, so no source tests would be needed for the inner tables.

Reply:
Alright, I simplified and removed the Table case where it's no longer necessary.